sway/list.c

```diff
@@ -20,25 +28,20 @@ void list_free(list_t *list) {
 }
 
 void list_add(list_t *list, void *item) {
-	if (list->length == list->capacity) {
-		list->capacity += 10;
-		list->items = realloc(list->items, sizeof(void*) * list->capacity);
-	}
+	list_resize(list);
 	list->items[list->length++] = item;
 }
 
 void list_insert(list_t *list, int index, void *item) {
-	// TODO: Implement this properly
-	if (list->length == list->capacity) {
-		list->capacity += 10;
-		list->items = realloc(list->items, sizeof(void*) * list->capacity);
-	}
-	list->items[list->length++] = item;
+	list_resize(list);
+	memmove(&list->items[index + 1], &list->items[index], sizeof(void*) * (list->length - index));
+	list->length++;
+	list->items[index] = item;
 }
 
 void list_del(list_t *list, int index) {
 	list->length--;
-	memmove(&list->items[index], &list->items[index + 1], sizeof(void*) * (list->capacity - index - 1));
+	memmove(&list->items[index], &list->items[index + 1], sizeof(void*) * (list->length - index));
```

Comment:
no problem, i wanted to try to implement the multikey handling with list_t but compiler complains about int->void* void*->int to much.
ill send another pr with it edited a bit later.